### PR TITLE
v4 integration

### PR DESCRIPTION
**Pull request by shireenf-ibm**
initial diff support (code + first successfully running bats test)

### CODE REVIEW

File: roxctl/connectivity-map/cta.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 	"path/filepath"
 
 	npguard "github.com/np-guard/netpol-analyzer/pkg/netpol/connlist"
+	npguardUtil "github.com/np-guard/netpol-analyzer/pkg/netpol/eval"
```
Comment (adisos) [inline roxctl/connectivity-map/cta.go]: I would rename `npguardUtil` to `eval`, and `npguard` to `connlist`

File: roxctl/connectivity-diff/netpol.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	// injected or constructed values
+	env environment.Environment
+}
+
+// Command defines the connectivity-diff command tree
+func Command(cliEnvironment environment.Environment) *cobra.Command {
+	diffNetpolCmd := &diffNetpolCommand{env: cliEnvironment}
+	c := &cobra.Command{
+		Use:   "connectivity-diff <folder-path1> <folder-path2>",
+		Short: "(Technology Preview) Report connectivity diff based on two directories of network policies and other resources.",
```
Comment (adisos) [inline roxctl/connectivity-diff/netpol.go]:
```suggestion
		Short: "(Technology Preview) Report connectivity diff based on two directories of network policies and workload resources YAML manifests.",
```

File: roxctl/connectivity-diff/netpol.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		Long: `Based on given two folders containing deployment, network policy and other networking resources YAMLs, will Report all differences in allowed connections between the directories. 
+		Will write to stdout if no output flags are provided.
```
Comment (adisos) [inline roxctl/connectivity-diff/netpol.go]:
```suggestion
		Long: `Based on given two folders containing Kubernetes workloads and network policy YAMLs, will report all differences in allowed connections between the directories. 
		Will write to stdout if no output flags are provided.
```

File: roxctl/connectivity-diff/cta.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	cmd.env.Logger().PrintfLn(connsDiff)
+}
+
+func (cmd *diffNetpolCommand) getDefaultFileName() string {
+	fileNamePrefix := defaultOutputFileNamePrefix + cmd.inputFolderPath1 + period
+	if cmd.outputFormat == "" {
+		return fileNamePrefix + defaultOutputFormat
+	}
+	return fileNamePrefix + cmd.outputFormat
+}
```
Comment (adisos) [inline roxctl/connectivity-diff/cta.go]: I would simplify the output file name to be just `connectivity_diff.txt` instead of involving input file names

**Review by adisos — CHANGES_REQUESTED** [review]
a few initial comments, still need to review the tests

File: roxctl/connectivity-map/cta.go (hunk 1 above)
Comment (adisos) [reply roxctl/connectivity-map/cta.go]: another option: `npgeval` and `npgconnlist`

File: roxctl/connectivity-diff/netpol_test.go [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+			outputFormat:          "docx",
+			errStringContainment:  true,
+			expectedAnalysisError: errors.New("docx output format is not supported."),
+		},
+		{
+			name:                  "happy path",
+			inputFolderPath1:      "testdata/netpol-analysis-example-minimal",
+			inputFolderPath2:      "testdata/netpol-diff-example-minimal",
+			expectedValidateError: nil,
+			expectedAnalysisError: nil,
```
Comment (adisos) [inline roxctl/connectivity-diff/netpol_test.go:86]: should also add expected output here?

Comment (shireenf-ibm) [reply roxctl/connectivity-diff/netpol_test.go:86]: we have other tests on same dirs with expected outputs (lines 115-141 ) (this test actually was added just to see that there are no errors )

File: roxctl/connectivity-diff/cta.go (hunk 4 above)
Comment (shireenf-ibm) [reply roxctl/connectivity-diff/cta.go]: done

File: roxctl/connectivity-map/cta.go (hunk 1 above)
Comment (shireenf-ibm) [reply roxctl/connectivity-map/cta.go]: done

**Pull request closed without merging**